[All Hosts](shared runtime): Update and move shared runtime articles for PPT support (#2356)
OfficeDev/office-js-docs-pr: docs/excel/show-hide-add-in.md

---
title: Show or hide the task pane of your Office Add-in

> description: 'Learn how to programmatically hide or show the user interface of an add-in while it runs continuously'

Reviewer: ```suggestion
description: 'Learn how to programmatically hide or show the user interface of an add-in while it runs continuously.'
```

}
```

The previous code assumes a scenario where there is an Excel worksheet named **CurrentQuarterSales**. The add-in will make the task pane visible whenever this worksheet is activated. The method `onCurrentQuarter` is a handler for the [Office.Worksheet.onActivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview&preserve-view=true#onactivated) event which has been registered for the worksheet.

You can also hide the task pane by calling the `Office.addin.hide()` function.

```javascript
function onCurrentQuarterDeactivated() {
    Office.addin.hide();
}
```

The previous code is a handler that is registered for the [Office.Worksheet.onDeactivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview&preserve-view=true#ondeactivated) event.

## Additional details on showing the task pane

> When you call `Office.addin.showAsTaskpane()`, Office will display in a task pane the page that you assigned to the resource ID (`resid`) for the task pane. This is the `resid` that you assigned to the `<SourceLocation>` of the `<Action xsi:type="ShowTaskpane">` in the manifest. (See [Configure your Office Add-in to use a shared runtime](configure-your-add-in-to-use-a-shared-runtime.md).)

Reviewer 2: This section has a couple complicated concepts that confused me while reading, so here is a potential rephrasing: 
```suggestion
When you call `Office.addin.showAsTaskpane()`, Office will display in a task pane the file that you assigned as the resource ID (`resid`) value of the task pane. This `resid` value can be assigned or changed by opening your **manifest.xml** file and locating `<SourceLocation>` inside the `<Action xsi:type="ShowTaskpane">` element.
(See [Configure your Office Add-in to use a shared runtime](configure-your-add-in-to-use-a-shared-runtime.md) for additional details.)
```

removeVisibilityModeHandler();
```

> The `onVisibilityModeChanged` method is asynchronous and returns a promise, which means that your code needs to await the promise before it can call the **deregister** handler.

Reviewer 2: ```suggestion
The `onVisibilityModeChanged` method is asynchronous and returns a promise, which means that your code needs to await the fulfillment of the promise before it can call the **deregister** handler.
```

```javascript

> //await the promise from onVisibilityModeChanged and assign
> //the returned deregister handler to removeVisibilityModeHandler.

Reviewer: ```suggestion
// await the promise from onVisibilityModeChanged and assign
// the returned deregister handler to removeVisibilityModeHandler.
```

    });
```

The deregister function is also asynchronous and returns a promise. So, if you have code that should not run until after the deregistration is complete, then you should await the promise returned by the deregister function.

```javascript

> //await the promise from the deregister handler before continuing

Reviewer: ```suggestion
// await the promise from the deregister handler before continuing
```
